compiler/stable_mir/src/mir/body.rs

```diff
@@ -2,10 +2,52 @@ use crate::ty::{AdtDef, ClosureDef, Const, CoroutineDef, GenericArgs, Movability
 use crate::Opaque;
 use crate::{ty::Ty, Span};
 
+/// The SMIR representation of a single function.
 #[derive(Clone, Debug)]
 pub struct Body {
     pub blocks: Vec<BasicBlock>,
+
+    /// Declarations of locals.
+    //
+    // The first local is the return value pointer, followed by `arg_count`
+    // locals for the function arguments, followed by any user-declared
+    // variables and temporaries.
     pub locals: LocalDecls,
+
+    // The number of arguments this function takes.
+    arg_count: usize,
+}
+
+impl Body {
+    /// Constructs a `Body`.
+    ///
+    /// A constructor is required to build a `Body` from outside the crate
+    /// because the `arg_count` field is private.
+    pub fn new(blocks: Vec<BasicBlock>, locals: LocalDecls, arg_count: usize) -> Self {
+        // If locals doesn't contain enough entries, it can lead to panics in
+        // `ret_local` and `arg_locals`.
+        assert!(
+            locals.len() >= arg_count + 1,
```

Comment:
```suggestion
            locals.len() > arg_count,
```
Same thing but simpler. :)

Reply:
Sure thing! I was originally thinking that `arg_count + 1` implied that locals must contain arg_count locals and 1 local more (for the return local), but the text I have in the assertion makes that point more clearly.